crates/zoon/src/style/width.rs

```diff
@@ -19,13 +21,31 @@ impl Default for WidthMode {
 }
 
 impl<'a> Width<'a> {
+    /// Define the width with pixels for an element.
+    /// # Example
+    /// ```no_run
+    /// use zoon::*;
+    ///
+    /// let button = Button::new().s(Width::new(50)).label("Click ne");
+    /// ```
     pub fn new(width: u32) -> Self {
         let mut this = Self::default();
         this.static_css_props.insert("width", px(width));
         this.width_mode = WidthMode::Exact;
         this
     }
 
+    /// Define the width with pixels depending of signal's state.
+    /// # Example
+    /// ```no_run
+    /// use zoon::*;
+    ///
+    /// let (is_hovered, hover_signal) = Mutable::new_and_signal(false);
+    /// let button = Button::new()
+    ///     .s(Width::with_signal(hover_signal.map_bool(|| 50, || 100)))
+    ///     .on_hovered_change(move |hover| is_hovered.set(hover))
+    ///     .label("hover me");
```

Comment:
```suggestion
    /// use zoon::{named_color::*, *};
    ///
    /// let container_width = Mutable::new(0);
    /// let container_width_broadcaster = Broadcaster::new(container_width.signal());
    ///
    /// let button = Row::new()
    ///     .s(Align::center())
    ///     .s(Font::new().color(GRAY_8))
    ///     .s(Background::new().color(BLUE_4))
    ///     .s(Borders::all(Border::new().color(BLUE_8).width(3)))
    ///     .s(Spacing::new(20))
    ///     .s(Width::with_signal(
    ///         container_width_broadcaster.signal().map(|width| {
    ///             if width > 1000 {
    ///                 return width / 2;
    ///             }
    ///             (f64::from(width) * 0.9) as u32
    ///         }),
    ///     ))
    ///     .item("Change the window size!")
    ///     .item(
    ///         El::new()
    ///             .s(Align::new().right())
    ///             .child_signal(container_width_broadcaster.signal()),
    ///     );
    ///
    /// let container = El::new()
    ///     .s(Width::fill())
    ///     .s(Height::fill())
    ///     .on_viewport_size_change(move |width, _| container_width.set_neq(width))
    ///     .child(button);
```
This new example is relatively complex but it demonstrates a real-world use case: a Rust alternative to CSS breakpoints. Also there shouldn't be problems like in the previous example (e.g. infinite loops).

Reply:
Thank you so much for this nice example :) I completely forgot about `responsiveness` :facepalm: